src/test/java/com/wikia/webdriver/testcases/discussions/UploadingImageTests.java

```diff
@@ -0,0 +1,264 @@
+    startReplyCreationMobile(page).uploadImage().clickSubmitButton();
+    page.waitForPageReload();
+    Assertion.assertTrue(page.findNewestReply().hasImage(), REPLY_IMAGE_VISIBLE);
+  }
+
+  @Test(groups = MOBILE)
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  @Execute(onWikia = MOBILE_COMMUNITY, asUser = User.USER_2)
+  public void userCannotUploadUnsupportedImageToTheirPostOnMobile() {
+    PostsListPage page = new PostsListPage().open();
+    BasePostsCreator postCreator = startPostCreationMobile(page);
+    String errorMsg = postCreator.uploadUnsupportedImage();
+    Assertion.assertStringContains(errorMsg, UNSUPPORTED_IMAGE_MSG);
+    postCreator.clickSubmitButton();
+    page.waitForPageReload();
+    Assertion.assertFalse(page.getPost().firstPostHasImage(), POST_UNSUPPORTED_IMAGE_NOT_VISIBLE);
+  }
+
+  @Test(groups = MOBILE)
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  @Execute(onWikia = MOBILE_COMMUNITY, asUser = User.USER_2)
+  public void userCannotUploadUnsupportedImageToTheirReplyOnMobile() {
+    PostDetailsPage page = new PostDetailsPage().open(setUp(MOBILE_COMMUNITY).getId());
+    BaseReplyCreator replyCreator = startReplyCreationMobile(page);
+    String errorMsg = replyCreator.uploadUnsupportedImage();
+    Assertion.assertStringContains(errorMsg, UNSUPPORTED_IMAGE_MSG);
+    replyCreator.clickSubmitButton();
+    page.waitForPageReload();
+    Assertion.assertFalse(page.findNewestReply().hasImage(), REPLY_UNSUPPORTED_IMAGE_NOT_VISIBLE);
+  }
+
+  @Test(groups = MOBILE)
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  @Execute(onWikia = MOBILE_COMMUNITY, asUser = User.USER_2)
+  public void userCanRemoveImagePreviewFromPostDraftOnMobile() {
+    PostsListPage page = new PostsListPage().open();
+    startPostCreationMobile(page).uploadImage().removeImage().clickSubmitButton();
+    page.waitForPageReload();
+    Assertion.assertFalse(page.getPost().firstPostHasImage(), POST_DELETED_IMAGE_NOT_VISIBLE);
+  }
+
+  @Test(groups = MOBILE)
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  @Execute(onWikia = MOBILE_COMMUNITY, asUser = User.USER_2)
+  public void userCanRemoveImagePreviewFromReplyDraftOnMobile() {
+    PostDetailsPage page = new PostDetailsPage().open(setUp(MOBILE_COMMUNITY).getId());
+    startReplyCreationMobile(page).uploadImage().removeImage().clickSubmitButton();
+    page.waitForPageReload();
+    Assertion.assertFalse(page.findNewestReply().hasImage(), REPLY_DELETED_IMAGE_NOT_VISIBLE);
+  }
+
+  @Test(groups = MOBILE, enabled = false)
+  @RelatedIssue(issueID = "IRIS-4896", comment = "To be implemented")
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  @Execute(onWikia = MOBILE_COMMUNITY, asUser = User.USER_2)
+  public void userCanOverwriteOpenGraphImageInExistingPostWithUploadedImageOnMobile() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1186)